src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -74,6 +74,32 @@ public static void ReadPrimitivesFail()
             Assert.Throws<JsonException>(() => JsonSerializer.Parse<int>(@""""""));
         }
 
+        [Theory]
+        [InlineData(typeof(bool))]
+        [InlineData(typeof(byte))]
+        [InlineData(typeof(char))]
+        [InlineData(typeof(DateTime))]
+        [InlineData(typeof(DateTimeOffset))]
+        [InlineData(typeof(decimal))]
+        [InlineData(typeof(double))]
+        [InlineData(typeof(JsonTokenType))]
+        [InlineData(typeof(Guid))]
+        [InlineData(typeof(short))]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(long))]
+        [InlineData(typeof(sbyte))]
+        [InlineData(typeof(float))]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(ushort))]
+        [InlineData(typeof(uint))]
+        [InlineData(typeof(ulong))]
+        public static void PrimitivesShouldFailWithArrayOrObjectAssignment(Type primitiveType)
+        {
+            // This test lines up with the built in JsonValueConverters
+            Assert.Throws<JsonException>(() => JsonSerializer.Parse(@"[]", primitiveType));
+            Assert.Throws<JsonException>(() => JsonSerializer.Parse(@"{}", primitiveType));
```

Comment:
It looks like these tests pass as-is, which indicates https://github.com/dotnet/corefx/issues/37807 is a valid issue. If we are throwing `JsonException` for other primitives, then I retract my comment [here](https://github.com/dotnet/corefx/issues/37807#issuecomment-494110078) about DateTime/Offset/Guid (since we should be consistent).